src/lib.rs

```diff
@@ -37,7 +36,7 @@ mod parse;
 
 #[derive(Debug)]
 pub enum Error {
-    Source(VecDeque<CompileError>),
+    Source(VecDeque<Box<dyn NewCompileError>>),
```

Comment:
This double-boxes the errors. I would rather see `VecDeque<dyn NewCompileError>`.

Reply:
This is not possible, `T` must implement `Sized`.